modules/passkey/test/userStories/passkeyCredentialCreation.spec.ts

```diff
@@ -0,0 +1,147 @@
+    const [user] = await ethers.getSigners()
+
+    const proxyFactory = await ethers.getContractAt(SafeProxyFactory.abi, SafeProxyFactory.address)
+    const singleton = await ethers.getContractAt(SafeL2.abi, SafeL2.address)
+    const fallbackHandler = await ethers.getContractAt(CompatibilityFallbackHandler.abi, CompatibilityFallbackHandler.address)
+    const signerFactory = await ethers.getContractAt('WebAuthnSignerFactory', WebAuthnSignerFactory.address)
+    const verifier = await ethers.getContractAt('IP256Verifier', FCLP256Verifier.address)
+    const verifierAddress = await verifier.getAddress()
+
+    const navigator = {
+      credentials: new WebAuthnCredentials(),
+    }
+
+    // Create the credentials for Passkey.
+    const credential = navigator.credentials.create({
+      publicKey: {
+        rp: {
+          name: 'Safe',
+          id: 'safe.global',
+        },
+        user: {
+          id: ethers.getBytes(ethers.id('chucknorris')),
+          name: 'chucknorris',
+          displayName: 'Chuck Norris',
+        },
+        challenge: ethers.toBeArray(Date.now()),
+        pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      },
+    })
+
+    // Get the publicKey from the credential and create the signer.
+    const publicKey = decodePublicKey(credential.response)
+    await signerFactory.createSigner(publicKey.x, publicKey.y, verifierAddress)
+    const signerAddress = await signerFactory.getSigner(publicKey.x, publicKey.y, verifierAddress)
+    const signer = await ethers.getContractAt('WebAuthnSigner', signerAddress)
+
+    // Deploy Safe with the WebAuthn signer as a single owner.
+    const singletonAddress = await singleton.getAddress()
+    const setupData = singleton.interface.encodeFunctionData('setup', [
+      [signerAddress],
+      1,
+      ethers.ZeroAddress,
+      '0x',
+      await fallbackHandler.getAddress(),
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ])
+    const safeAddress = await proxyFactory.createProxyWithNonce.staticCall(singletonAddress, setupData, 0)
+    await proxyFactory.createProxyWithNonce(singletonAddress, setupData, 0)
+    const safe = await ethers.getContractAt([...SafeL2.abi, ...CompatibilityFallbackHandler.abi], safeAddress)
+
+    return {
+      user,
+      safe,
+      safeAddress,
+      signer,
+      navigator,
+      credential,
+    }
+  })
+
+  it('should be possible to execute a transaction signed by passkey', async () => {
+    const { user, safe, safeAddress, signer, navigator, credential } = await setupTests()
+
+    // Send 1 ETH to the Safe
+    await user.sendTransaction({ to: safe, value: ethers.parseEther('1') })
+
+    // Define transaction to be signed.
+    const nonce = await safe.nonce()
+    const randomAddress = ethers.getAddress(ethers.hexlify(ethers.randomBytes(20)))
+    const safeTx = buildSafeTransaction({ to: randomAddress, value: 1n, nonce: nonce })
+    const safeTxData = await safe.encodeTransactionData.staticCall(
```

Comment:
Why on-chain? The general rule of thumb is to do things off-chain to save the RPC from some load.